x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -3234,6 +3235,36 @@ public void testEmptyMappingIndex() {
         assertThat(Expressions.names(local.output()), contains(NO_FIELDS.get(0).name(), "x", "language_code", "language_name"));
     }
 
+    public void testPlanSanityCheck() throws Exception {
```

Comment:
I think we should also add a test for the case where a logical plan node in the middle of the plan has invalid references, not only the top node.